src/back/backend_handler.rs

```diff
@@ -33,6 +33,7 @@ pub struct BackEndHandler {
     provider: Arc<dyn Provide + Send + Sync>,
     #[derivative(Debug = "ignore")]
     converter: Box<dyn Convert + Send + Sync>,
+    provider_name: String,
```

Comment:
Why do you need the `provider_name` here in the backend handler? For a future step of the implementation?

Reply:
Think I thought it might be required to link the backend handler to the provider but I can see there is already a reference to the provider object itself. Have removed this for now.